docs/src/environments.md

```diff
@@ -107,3 +112,146 @@ Otherwise, it will resolve the latest versions of the dependencies compatible wi
+  Installing known registries into `/tmp/juliatmp`
+      Status `/tmp/juliatmp/environments/v1.6/Project.toml` (empty project)
+
+(@v1.6) pkg> add Example
+    Updating registry at `/tmp/juliatmp/registries/General.toml`
+   Resolving package versions...
+   Installed Example ─ v0.5.3
+    Updating `/tmp/juliatmp/environments/v1.6/Project.toml`
+  [7876af07] + Example v0.5.3
+    Updating `/tmp/juliatmp/environments/v1.6/Manifest.toml`
+  [7876af07] + Example v0.5.3
+Precompiling project...
+  1 dependency successfully precompiled in 0 seconds
+```
+
+This way, anything you did with Pkg will be stored in the temporary directory `/tmp/juliatmp` so you
+do not need to worry that it will break your common workflow.
+
+## [shared environment](@id shared-environment)
+
+For package development, it is a good practice to maintain a as small as possible set of dependencies in independent project folder.
+However, from the users' perspective, it is often the case that we want to have convenient tools installed globally so that we don't
+need to switch between different environments too often. The default version-specific global environment is `@v#.#` stored
+in `~/.julia/environments` folder.
+
+For instance, [BenchmarkTools.jl](https://github.com/JuliaCI/BenchmarkTools.jl) provides a set of utilities we need to properly
+benchmark the function runtime and memory usage.
+
+```julia-repl
+(@v1.6) pkg> st
+      Status `~/.julia/environments/v1.6/Project.toml`
+  [6e4b80f9] BenchmarkTools v1.3.1
+
+(@v1.6) pkg> activate MyProject
+  Activating new environment at `~/MyProject/Project.toml`
+
+julia> using BenchmarkTools
+
+```
+
+Loading `BenchmarkTools` in MyProject environment works even if this package is not recorded in `~/MyProject/Project.toml`. This is
+because Julia maintains a environments stack via the `LOAD_PATH` variable, see also the [environment stacks](https://docs.julialang.org/en/v1/manual/code-loading/#Environment-stacks) section in the julia manual.
+
+```julia-repl
+julia> LOAD_PATH
+3-element Vector{String}:
+ "@"
+ "@v#.#"
+ "@stdlib"
+```
+
+Briefly speaking, Julia will try to load package `BenchmarkTools` in `@` environment, which is the current activated project.
+If not found then try the version-specific global environment `@v#.#`, which is stored in `~/.julia/environments/v1.6` for Julia 1.6.
+In the previous case, Julia found `BenchmarkTools` in `@v#.#`, i.e., `~/.julia/environments/v1.6/Project.toml`,
+thus it successfully loads the package.
+
+A convenient command to create custom shared environment is to prefix the name with `@`:
+
+```julia-repl
+(@v1.6) pkg> activate @mytoolbox # require Julia 1.4 or later
+  Activating new project at `~/.julia/environments/mytoolbox`
+```
+
+This provides a quick way to activate the project environment stored in `~/.julia/environments` folder.
+
+### Shared environment caveat: the reproducibility issue
+
+Using shared environment can cause reproducibility issues. For instance,
```

Comment:
```suggestion
!!! warning Shared environments and reproducibility
    Because shared environments allow you to load packages without specifying them in a particular environment's
    `Project.toml` and `Manifest.toml`, you may encounter problems with reproducibility.
```

Reply:
I'm not sure if it's appropriate to use `!!! ` for a long thread like this, so I kept the `###` title here.